spec/support/concerns/project_updater_shared_examples.rb

```diff
@@ -0,0 +1,71 @@
+shared_examples 'a ProjectUpdater' do
+  let(:project) { subject.project }
+  let(:create_subject) {
+    subject.save
+    subject.reload
+  }
+
+  it {
+    is_expected.to respond_to(:update_project_etag)
+    is_expected.to callback(:update_project_etag).after(:save)
+    is_expected.to callback(:update_project_etag).after(:destroy)
+  }
+
+  describe '#update_project_etag' do
+    let!(:original_project_etag) { project.etag }
+    it {
+      expect(create_subject).to be_truthy
+      expect(change_subject).to be_truthy
+      subject.update_project_etag
+      project.reload
+      subject.reload
+      last_project_audit = project.audits.last
+      last_subject_audit = subject.audits.last
+      expect(project.etag).not_to eq(original_project_etag)
+
+      expected_comment = last_subject_audit.comment ? last_subject_audit.comment.merge({raised_by_audit: last_subject_audit.id}) : {raised_by_audit: last_subject_audit.id}
```

Comment:
Logic like this is "code smell". Or is it "test smell"? 😛 We can control the subject's audit comment with `subject.audit_comment = {'foo' => 'bar'}` or `subject.audit_comment = nil`. Maybe use the former as the default and then test the latter as a separate context. If you use a let variable, you can easily override it for your nil audit_comment context.

Reply:
thanks. I had forgotten about audit_comment